src/xdp/xsk.c

```diff
@@ -4134,13 +4171,12 @@ XskNotify(
             &Xsk->IoWaitEvent, UserRequest, UserMode, FALSE,
             (TimeoutMilliseconds == INFINITE) ? NULL : &Timeout);
 
-    //
-    // Clean up the wait context.
-    //
     KeAcquireSpinLock(&Xsk->Lock, &OldIrql);
-    Xsk->IoWaitFlags = 0;
+    if (Xsk->IoWaitFlags == XSK_NOTIFY_FLAG_WAKE) {
+        Xsk->IoWaitFlags = 0;
+        Status = STATUS_CANCELLED;
```

Comment:
Should we `goto Exit` after this or still allow the `XskQueryReadyIo` to overwrite the result to `STATUS_SUCCESS`?

Reply:
I'm leaning towards `goto Exit`.

Reply:
Hmm, it depends on our terminology. IMO the flag itself should be called XSK_NOTIFY_FLAG_CANCEL_WAIT, and then the 1:1 combination of cancelation here and returning "wait canceled" on the canceling thread makes sense.

If we keep "FLAG_WAKE" then returning STATUS_CANCELLED makes quite a bit less sense, both literally (mismatch of terms) and in the inconsistency that a wait can time out (above) and yet return a successful status if the wait conditions were eventually satisfied below.

So my suggestion is we rename the flag to XSK_NOTIFY_FLAG_CANCEL_WAIT and then always return STATUS_CANCELLED when the cancel path occurs.